### ISSUE TITLE
support for saving datasets

### ISSUE THREAD

**Issue by amrsharaf** [pytorch/text#323]
Currently torchtext doesn't seem to support saving datasets in an easy way. Pickle doesn't also seem to work due to the existence of generators and lambda functions in the Dataset / Field objects, would it be possible to provide and API for saving datasets? This is particularly helpful for loading pre-processed datasets where the pre-processing / vocabulary creation takes a long time that can be skipped by doing once and then loaded a saved pre-processed copy of the dataset.

**Comment by nh007cs** [comment]
Looking forward to that! That's helpful!

**Comment by domaala** [comment]
This has been a hot topic for quite a while, see #140, #253, and #270. It's work in progress.

**Comment by edult** [comment]
Has this been resolved?

**Comment by risingdhxs** [comment]
Also looking forward to this support function, as calling datasets.IWSLT.splits(...) takes about an hour on my computer...
On that topic, is there a way to speed up this process? From what I observe only one core was running this process. Utilizing multiple cores should speed it up considerably

**Comment by zhangguanheng66** [comment]
@domaala just wondering if there is any progress for this issue. Thanks.

**Comment by plkmo** [comment]
Me too interested in this saving support

**Comment by dainis-boumber** [comment]
Yes, this would have saved me some headache. I ended up doing a piecemeal solution that involved some pandas, some basic python, some torchtext. Making a more conformant purely torchtext solution over this and next weekend, if I have time and there is still interest, I  can try to work on a mergeable solution sometime next month.

**Comment by zhangguanheng66** [comment]
@dainis-boumber. I'm wondering if you are interested to share your work via a PR. Thanks.
